Ginger/MSAccessDB/MSAccessDBCon.cs

```diff
@@ -0,0 +1,352 @@
+                }
+                else { strProvider = "Provider=Microsoft.ACE.OLEDB.12.0;"; }
+
+                connStr = strProvider + connStr;
+            }
+            return connStr;
+        }
+
+        public string GetSingleValue(string Table, string Column, string Where)
+        {
+            string sql = "SELECT {0} FROM {1} WHERE {2}";
+            sql = String.Format(sql, Column, Table, Where);
+            String rc = null;
+            DbDataReader reader = null;
+            if (MakeSureConnectionIsOpen())
+            {
+                try
+                {
+                    DbCommand command = conn.CreateCommand();
+                    command.CommandText = sql;
+                    command.CommandType = CommandType.Text;
+
+                    // Retrieve the data.
+                    reader = command.ExecuteReader();
+                    while (reader.Read())
+                    {
+                        rc = reader[0].ToString();
+                        break; // We read only first row
+                    }
+                }
+                catch (Exception e)
+                {
+                    throw e;
+                }
+                finally
+                {
+                    reader.Close();
+                }
+            }
+            return rc;
+        }
+
+        public List<string> GetTablesColumns(string table)
+        {
+            DbDataReader reader = null;
+            List<string> rc = new List<string>() { "" };
+                try
+                {
+                    DbCommand command = conn.CreateCommand();
+                    // Do select with zero records
+                    command.CommandText = "select * from " + table + " where 1 = 0";
+                    command.CommandType = CommandType.Text;
+
+                    reader = command.ExecuteReader();
+                    // Get the schema and read the cols
+                    DataTable schemaTable = reader.GetSchemaTable();
+                    foreach (DataRow row in schemaTable.Rows)
+                    {
+                        string ColName = (string)row[0];
+                        rc.Add(ColName);
+                    }
+                }
+                catch (Exception e)
+                {
+                    Reporter.ToLog(eLogLevel.ERROR, "", e);
+                    throw (e);
+                }
+                finally
+                {
+                    reader.Close();
+                }
+            return rc;
+        }
+
+        public bool MakeSureConnectionIsOpen()
+        {
+            Boolean isCoonected = true;
+
+            if ((conn == null) || (conn.State != ConnectionState.Open))
+            {
+                isCoonected = OpenConnection(KeyvalParamatersList);
+            }
+            //make sure that the connection was not refused by the server               
+            TimeSpan timeDiff = DateTime.Now - LastConnectionUsedTime;
```

Comment:
I don't think we need the below check and why 5 mins?

Reply:
removed